modules/filecheck/init.go

```diff
@@ -0,0 +1,35 @@
+package filecheck
+
+import (
+	"errors"
+
+	"github.com/netdata/go-orchestrator/module"
+)
+
```

Comment:
I want to confirm one conclusion, for the next two `methods`, you are copying the whole object instead to give as argument. Are you doing this because they are only called during the initialization ?

Reply:
i'm not copying anyting, i'm importing that package

Reply:
Sorry, I did not express well. I was talking about [this situation](https://tour.golang.org/methods/4). The methods does not have pointers, so you are copying the whole data before to do the tests, or am I missing something?

Reply:
can you show the exact part of the code you are talking about? you highlighted `import`...

Reply:
ok, i think i get it, you are talking about `Filecheck` methods in the `init.go` file.

A function/method declaration should tell you a story - what it does, what it want, what it produces.

Ptr receiver means you can modify the value to which the receiver points. So if it is so i would assume you do so.

In that situation it is not ptr receiver because i dont need to change anything and you can say that just by the method declaration.

The rule is:
 - if you need to modify the value to which the receiver points - ptr rcvr
 - if you dont - value rcvr
 - if in doubts - ptr rcvr

> copying the whole data before

I suggest to not take it into account atm, there are cases when you want to have ptr rcvr just for optimization, but this case is **far** from it.

Copying is everywhere (it is either value copy or ptr copy), it is ok until it is a lot of heap allocations/ memory leak.

So for now just follow the rules i've written above.